### PR TITLE
Make the left sidebar resizable

### PR DESCRIPTION
**Pull request by morgandruesne**
## What does this PR do ?
This PR make the left sidebar resizable in data page

### How should this be manually tested?

Be sure to have an instances of Kuzzle. Create different indexes and collection with different name size, login and try to resize the left sidebar

ezgif com-gif-maker

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,9 +40,9 @@
     "babel-polyfill": "^6.26.0",
     "bootstrap-vue": "^2.13.0",
     "core-js": "^3.4.3",
+    "direct-vuex": "^0.9.3",
```
Comment (Leodau) [inline package.json:43]: Why is this dependency introduced?

File: src/components/Data/Leftnav/IndexBranch.vue [hunk 2]
```diff
@@ -197,10 +195,17 @@ export default {
 </script>
 
 <style scoped lang="scss">
+.IndexBranch {
```
Comment (Leodau) [inline src/components/Data/Leftnav/IndexBranch.vue:198]: Does this class name follow this project naming convention? and is it necessary for this component?

File: package.json (hunk 1 above)
Comment (morgandruesne) [reply package.json:43]: She was already there but line 45, I think I move it inadvertently.

File: src/components/Data/Leftnav/IndexBranch.vue (hunk 2 above)
Comment (morgandruesne) [reply src/components/Data/Leftnav/IndexBranch.vue:198]: The name of the class already existed but was not used. So I think he's respecting the naming convention. There's a more appropriate name?

Comment (xbill82) [reply src/components/Data/Leftnav/IndexBranch.vue:198]: This name sucks. My fault. Fustigate Luca from 2016, please.

Comment (Leodau) [reply src/components/Data/Leftnav/IndexBranch.vue:198]: If this class is not used, I believe it shouldn't be included in this PR (https://deviq.com/boy-scout-rule/).

Comment (morgandruesne) [reply src/components/Data/Leftnav/IndexBranch.vue:198]: She wasn't. But she is now. Any suggestions for the name?

**Comment by morgandruesne** [comment]
Yeap, I'll add this !

**Comment by morgandruesne** [comment]
So i add a visual hint for the user to see that they can grab the edge of the sidebar and resize it, as you advised ! :smile: 
Screenshot from 2020-05-22 11-02-35

File: src/components/Data/Layout.vue [hunk 3]
```diff
@@ -71,4 +71,27 @@ export default {
   overflow: auto;
   padding: $content-gutter;
 }
+
+.Custom-resizer > .multipane-resizer {
+  margin: 0; left: 0;
+  position: relative;
+  padding: 3px;
+  border: 1px solid #ccc;
```
Comment (xbill82) [inline src/components/Data/Layout.vue:79]: Would be great to move the box-shadow from the wrapper to the resizer

```suggestion
  border: 1px solid #ccc;
  box-shadow: 0px 0px 5px 0px rgba(112, 112, 112, 1);
```

**Pull request merged**